### PR TITLE
Integrate Background effects from Longpipe effects SDK (try #2)

### PR DESCRIPTION
**Pull request by sb2702**
Adding in (hopefully) sharper and more performant virtual backgrounds from [Longpipe SDK](https://github.com/sb2702/longpipe)

Adding it behind a flag, setting USE_LONGPIPE to false should go back to the old tflite behavior

New:
Screenshot from 2026-05-07 11-41-39

Old:

Screenshot from 2026-05-07 11-42-18

It's still pretty early / in "MVP" stage right now, but even as-is, it should have better quality and be more performant than TFLite/Mediapipe.

Working on improvements to reduce flickering right now, future changes should just be a version bump

In the PR, to make a commit, there was something weird where even just adding a single line space would change the entire file, I think it had something to do with line spaces (crlf, lf) or whatever, I isolated the new line changes to the first commit, and the virtual background changes to the second commit

### CODE REVIEW

**Comment by steveseguin** [comment]
Hey @sb2702,

Thank you for the PR.

I ran into a snag when trying to implement things.

The main issue is that I need to be able to host file dependencies within VDO.Ninja itself, such as in the vdo.ninja/thirdparty folder or on a fileserver I control. I can't host files using your domain, for privacy and security reasons, and I try to avoid hosting on npm.

While the SDK code is MIT, the `.bin` model weights are distributed separately without a clear license. They will also need to be redistributable under permissive terms such as MIT, Apache-2.0, or BSD.

At a minimum, an explicit license for the model weights is needed, such as adding a WEIGHTS_LICENSE or MODEL_LICENSE file to the Longpipe repo/CDN package.

It would be nice to have the following also:
- confirmation that redistribution/self-hosting inside another MIT-compatible open-source project is allowed,
- whether any third-party/pretrained weights or datasets impose non-commercial or attribution restrictions.
- training data/source provenance at a high level

It looks pretty good on first pass, so I'm keen to get it deployed here and tested more.

-steve

**Comment by sb2702** [comment]
> Hey @sb2702,
> 
> Thank you for the PR.
> 
> I ran into a snag when trying to implement things.
> 
> The main issue is that I need to be able to host file dependencies within VDO.Ninja itself, such as in the vdo.ninja/thirdparty folder or on a fileserver I control. I can't host files using your domain, for privacy and security reasons, and I try to avoid hosting on npm.
> 
> While the SDK code is MIT, the `.bin` model weights are distributed separately without a clear license. They will also need to be redistributable under permissive terms such as MIT, Apache-2.0, or BSD.
> 
> At a minimum, an explicit license for the model weights is needed, such as adding a WEIGHTS_LICENSE or MODEL_LICENSE file to the Longpipe repo/CDN package.
> 
> It would be nice to have the following also:
> 
> * confirmation that redistribution/self-hosting inside another MIT-compatible open-source project is allowed,
> * whether any third-party/pretrained weights or datasets impose non-commercial or attribution restrictions.
> * training data/source provenance at a high level
> 
> It looks pretty good on first pass, so I'm keen to get it deployed here and tested more.
> 
> -steve

Hey Steve,

You can download the weights [here](https://cdn.longpipe.dev/models/v/0.0.1/index.html), I've also explicitly added a license for the weights [here](https://cdn.longpipe.dev/models/v/0.0.1/WEIGHTS_LICENSE)

To self host, just place all the weights in some publicly accessible file server, and then you can just add the following parameter:
```
new EffectsPipeline(stream, {
  weightsBaseUrl:   // Wherever you want to host
})
```

Or, if you could just host them, and tell me the base url path and I can update it in the PR?

Finally, getting to your other questions:

**confirmation that redistribution/self-hosting inside another MIT-compatible open-source project is allowed,**

- Absolutely

**whether any third-party/pretrained weights or datasets impose non-commercial or attribution restrictions**

- The weights themselves don't have any restrictions, I'm releasing them under the MIT license. Datasets are open source too

**training data/source provenance at a high level**

- I used the following datasets ([P3M-10k](https://github.com/JizhiziLi/P3M), [AISegment Human MAtting](https://www.kaggle.com/datasets/laurentmih/aisegmentcom-matting-human-datasets)) which are both released under MIT [[1](https://github.com/JizhiziLi/P3M)][[2](https://github.com/aisegmentcn/matting_human_datasets/blob/master/LICENSE)]

- The model uses a pre-trained EfficientNetLite backbone, the weights for which are released under [Apache 2.0](https://github.com/RangiLyu/EfficientNet-Lite/blob/main/LICENSE)

**Comment by sb2702** [comment]
Bumping the version with 0.0.4, which has higher definition masks and improved temporal stability

https://cdn.longpipe.dev/models/v/0.0.4/index.html

**Comment by steveseguin** [comment]
deployed for testing to https://vdo.ninja/alpha/

thank you
